dashboard/app/models/pd/application/teacher1920_application.rb

```diff
@@ -346,6 +348,9 @@ def allow_sending_principal_email?
       # Only if we haven't gotten a principal response yet.
       return false if response
 
+      # Only if we've sent at least one principal approval email before.
+      return false unless last_principal_approval_email_created_at
```

Comment:
At a glance, this logic reads an awful lot like it's used every time we try to send a principal approval email - which would mean we can never send one for a new application, after this check is added.

Confirming my understanding that this change depends on us sending a principal approval email here that does _not_ go through this check:

https://github.com/code-dot-org/code-dot-org/blob/a7f8912921d8f7e8f879c591ddfec73cc0b7f8cc/dashboard/app/models/pd/application/teacher1920_application.rb#L695

I wonder if it's worth renaming some methods in here to make it clear that they are only used in the follow-up or manual case, e.g. `allow_resending_principal_email?`

Reply:
Oh, great catch, because there is a relatively new [check](https://github.com/code-dot-org/code-dot-org/blob/fed11c8d253171c6016a246ae25aee7601d47831/dashboard/app/controllers/api/v1/pd/application/teacher_applications_controller.rb#L15) I added to the controller for a send email button which would be affected by this change.  I will plumb through a variation.

Reply:
Actually, I'll just move the check out of `allow_sending_principal_email?` and into `allow_sending_principal_approval_teacher_reminder_email?`